docs/content/management-bundles.md

```diff
@@ -0,0 +1,812 @@
+
+#### Signature Plugin
+
+OPA supports the option to implement your own bundle signing and verification logic. This will be unnecessary
+for most and is intended for advanced use cases, such as leveraging key-related services from cloud providers.
+To implement your own signing and verification logic, you'll need to [extend OPA](../extensions). Here is
+[an example](https://github.com/open-policy-agent/contrib/tree/master/custom_bundle_signing) to get you started.
+
+When registering custom signing and verification plugins, you will need to register the Signer and the Verifier
+under the same plugin key, because the plugin key is stored in the signed bundle and informs OPA which Verifier
+is capable of verifying the bundle, e.g.
+
+```go
+bundle.RegisterSigner("custom", &CustomSigner{})
+bundle.RegisterVerifier("custom", &CustomVerifier{})
+```
+
+## Implementations
+
+The Bundle API is simple. Most HTTP servers capable of serving static files will do. While not strictly required in all deployments, it is also good if the implementation supports:
+
+* HTTP caching using the [ETag header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag). This keeps OPA from having to download a bundle unless the bundle's content have changes.
+* Authentication. When exposing a bundle at a remote endpoint, it is often desirable to protect the data by requiring all requests to the endpoint to be authenticated.
+
+This document lists some of the more common HTTP servers suitable as bundle servers, along with instructions for how to set them up as such.
+
+### Amazon S3
+
+#### OPA Bundle Support
+
+| Feature | Supported |
+|---------|-----------|
+| Caching headers | Yes |
+| Authentication methods | [AWS Signature](https://www.openpolicyagent.org/docs/latest/configuration/#aws-signature) |
+
+#### Setup Instructions
+
+1. Search for "S3" and on the "Buckets" page, click "Create bucket".
+2. Fill in the form according to your preferences (name, region, etc).
+3. Either choose "Block all public access" for internal systems, or unmark the checkbox for that to allow external (authenticated) requests.
+4. You can now upload your bundle to the bucket. If you try to download it right away you'll notice that by default you're unauthorized to do so.
+5. To allow anyone to read the bundle, click on it and select "Make public" from the "Object actions" dropdown menu. If not, proceed to configure authentication.
+
+#### Authentication
+
+Authentication can be configured to either use the credentials of a service account stored in the environment, or to use credentials fetched from the AWS metadata API. The latter is only available from services running inside of AWS (on EC2 or ECS).
+
+Both methods are going to need a policy for either the service account or the IAM role, so when that is mentioned in the steps for either method you may refer to the example below.
+
+**Example IAM policy**
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
```

Comment:
Nit: We don't need ListBucket and PutObject, do we? If we just have one IAM policy for simplicity (so that users can list/upload objects and OPAs can get objects), that's fine but let's include a note or something.

Reply:
Yeah I didn't want to get too deep into explaining the differences between IAM policies, bucket policies and whether there should be different user/roles for uploads and downloads.. but you're right, I'll add a note saying that the OPA client need only GetObject.